cli/install/template.go

```diff
@@ -177,9 +177,9 @@ spec:
         args:
         - "proxy-api"
         - "-addr=:8086"
-        - "-metrics-addr=:9996"
-        - "-destination-addr=:8089"
-        - "-telemetry-addr=:8087"
+        - "-metrics-addr=127.0.0.1:9996"
+        - "-destination-addr=127.0.0.1:8089"
+        - "-telemetry-addr=127.0.0.1:8087"
```

Comment:
I don't think any of these even need to be configurable now: we can and should just hard-code `127.0.0.1:9996`, `127.0.0.1:8089`, etc. into the Go code and remove the flags. Then we can write tests in each server that start the server up, bind the port on the non-loopback addresses, make a request to localhost, and verify the request succeeds.

Otherwise, we should have a test in conduit-tests that verifies, at least, that the controller pod doesn't accept connections on these ports from outside the pod.

Reply:
I disagree: these addresses (including ip and port) are exactly the type of thing that should remain configurable.  This allows us to easily change the way they are hooked up.  For example, when we split these into separate pods, they'll need to start listening on non-localhost interfaces again.  Having this be configurable gives us the flexibility to run these components in a variety of configurations.

I've tested manually that the controller pod does not accept connections on 8087, 8088, and 8089 on the pod ip.

Reply:
> when we split these into separate pods

There is no plan to split them into separate pods. But, OK, we can figure out that part later.

Reply:
I agree with @adleong -- these should be configurable, and we should support multiple configurations but default to listening only on localhost. This is necessary for the docker-compose environment specifically, but will be generally useful in the event that we need to reconfigure going forward. Locking the code directly to today's deployment topology will make it harder to change later.

Reply:
Some other food for thought -- most (all?) of the `*addr` flags in this configuration file match the defaults in the go code. If we're worried about folks accidentally overriding the defaults, we might want to consider removing all of the flags here and relying directly on the defaults. I could go either way on this honestly, and I could see upsides and downsides of either approach.